gamedev/part3.md

```diff
@@ -0,0 +1,194 @@
+# Game Dev
+## Part 3: Different kinds of blocks using inheritance
+<a href="README.md#instructions">Back</a>
+
+### Intro to classes and objects
+
+A lot of our code so far has been involving the character sprite and ground sprites directly. However, there's a lot that's similar between them, and there's a lot that will be similar if we decide to create other kinds of blocks too, such as a ground block that bounces you extra high. All these different kinds of objects in the game can be considered parts of general categories of game objects, similar to how a square is a more specific kind of rectangle, which is a more specific kind of polygon, and there are other specific types of polygons such as triangles.
+
+In programming we refer to these categories as **classes**, and a more specific version of a category is referred to as a **subclass or child class** of the **parent or super class.**
+
+In Javascript, classes are implemented using **functions and prototypes.** A **class function** is used to initialize an instance of a class, and a **class prototype** defines what methods instances of the class have. A **base class** refers to the top-level category that other classes are all children of at some level. The polygon example might look like this:
```

Comment:
A lot of terminology here. Maybe have a definition for each of those terms and explain them a bit more. What is a class function?